src/sinks/datadog/logs/sink.rs

```diff
@@ -100,63 +99,60 @@ pub struct LogSink<S> {
 fn normalize_event(event: &mut Event) {
     let log = event.as_mut_log();
 
-    if let Some(message_path) = log.message_path().cloned().as_ref() {
-        log.rename_key(message_path, event_path!("message"));
-    }
-
-    if let Some(host_path) = log.host_path().cloned().as_ref() {
-        log.rename_key(host_path, event_path!("hostname"));
-    }
+    // Upstream Sources may have semantically defined Datadog reserved attributes outside of their
+    // expected location by DD logs intake (root of the event). Move them if needed.
+    DD_RESERVED_SEMANTIC_ATTRS.iter().for_each(|meaning| {
```

Comment:
nit: Since this is not at the end of a chain of iterators, it is more idiomatic to just use `for`:
```suggestion
    for meaning in &DD_RESERVED_SEMANTIC_ATTRS {
```

Reply:
FWIW I also prefer plain `for`